addons/web/models/models.py

```diff
@@ -212,34 +218,118 @@ def fields_view_get(self, view_id=None, view_type='form', toolbar=False, submenu
+    def search_panel_select_range(self, field_name, **kwargs):
         """
-        Return possible values of the field field_name (case select="one")
-        and the parent field (if any) used to hierarchize them.
+        Return possible values of the field field_name (case select="one"),
+        possibly with counters and the parent field (if any) used to hierarchize them.
 
-        :param field_name: the name of a many2one category field
+        :param field_name: the name of a field;
+            of type many2one or selection.
+        :param search_domain: base domain of search
+        :param disable_counters: whether to count records by value
         :return: {
             'parent_field': parent field on the comodel of field, or False
             'values': array of dictionaries containing some info on the records
                         available on the comodel of the field 'field_name'.
-                        The display name (and possibly parent_field) are fetched.
+                        The display name, the count (how many records with that value)
+                        and possibly parent_field are fetched.
         }
         """
         field = self._fields[field_name]
-        supported_types = ['many2one']
+        supported_types = ['many2one', 'selection']
         if field.type not in supported_types:
             raise UserError(_('Only types %(supported_types)s are supported for category (found type %(field_type)s)') % ({
                             'supported_types': supported_types, 'field_type': field.type}))
 
-        Comodel = self.env[field.comodel_name]
-        fields = ['display_name']
-        parent_name = Comodel._parent_name if Comodel._parent_name in Comodel._fields else False
-        if parent_name:
-            fields.append(parent_name)
-        return {
-            'parent_field': parent_name,
-            'values': Comodel.with_context(hierarchical_naming=False).search_read([], fields),
-        }
+        search_domain = kwargs.get('search_domain', [])
+        disable_counters = kwargs.get('disable_counters', False)
+
+        # retrieve all possible values, and return them with their label and counter
+        if field.type == 'many2one':
+            # extract the id from a many2one value
+            def get_id(value):
+                return value and value[0]
+
+            # determine counters
+            counters = defaultdict(int)
+            # comodel_ids = []
+            if not disable_counters:
+                groups = self.read_group(search_domain, [field_name], [field_name])
+                for group in groups:
+                    record_id = get_id(group[field_name])
+                    counters[record_id] = group[field_name + '_count']
+                    # if record_id:
+                    #     comodel_ids.append(record_id)
+
+            # determine result
+            result = lazymapping(lambda id: {
+                'id': id,
+                'count': 0,
```

Comment:
why not counters[id] and adapt code below?

Reply:
'count': counters[id] I mean

Reply:
This would avoid loop line 303 in case disable_counters is false and parent_name is not set